### ISSUE TITLE
Cannot import activities between two courses using the Grid format

### ISSUE THREAD

**Issue by jboulen** [gjb2048/moodle-format_grid#169]
**environment :**
moodle v4.0.4
grid 400.1.1
debian 11 (php 7.4 / mysql)

**steps to reproduce :**
1. create two courses and use grid format
2. define images on each sections
3. create an activity in the first course
4. go to the second course and use "import" or "course reuse" in course administration
5. select first course, and then "jump to final step"
6. an error should occur

![grid-error](https://user-images.githubusercontent.com/5418227/192817313-ed390de6-dcb9-458f-ada6-ede458823aee.png)
![grid-debug_info](https://user-images.githubusercontent.com/5418227/192817307-cbf9f0e4-9822-4957-a6a1-e7fe4217ee6b.png)
![grid-stack_trace](https://user-images.githubusercontent.com/5418227/192817297-329d2fea-191d-4029-859a-de54a8ac3204.png)

**workaround :**
We can add [this line](https://github.com/gjb2048/moodle-format_grid/blob/df490fb8ce8c63d96c7b3bf6cb50416f33649768/backup/moodle2/restore_format_grid_plugin.class.php#L234) in a try/catch to avoid the error. The import works, but maybe there is a side effect to ignore this exception...

**Comment by gjb2048** [comment]
@jboulen Q - on your system is '3' the course id of the new course?

**Comment by gjb2048** [comment]
@jboulen Interesting, thanks.

**Comment by jboulen** [comment]
Without this lines, the issue does not happen :
![grid-success](https://user-images.githubusercontent.com/5418227/192826036-50a77d41-3971-447a-ae91-0d6f1fc6535e.png)

**Comment by gjb2048** [comment]
Ok, solution breaks course duplication.

**Comment by gjb2048** [comment]
@jboulen Please test https://github.com/gjb2048/moodle-format_grid/commit/4769e4b9e48ca685a96961c311ea3ed06b5ae7bc - you can do a drop in replacement for https://raw.githubusercontent.com/gjb2048/moodle-format_grid/4769e4b9e48ca685a96961c311ea3ed06b5ae7bc/backup/moodle2/restore_format_grid_plugin.class.php.

**Comment by jboulen** [comment]
> @jboulen And does the image show please?

The original image for the second course has been preserved. However, when you edit section, there are two images in _section image_ field (the original image and image used by the first course).

**Comment by jboulen** [comment]
> [4769e4b](https://github.com/gjb2048/moodle-format_grid/commit/4769e4b9e48ca685a96961c311ea3ed06b5ae7bc)

It works. Activities has been correctly imported and there is only one image in _section image_ field when you edit section.

The section image in second course has been preserved. I think this is the desired behavior. It would probably be strange if the first course image was imported ?

Thank you very much.

**Comment by gjb2048** [comment]
@jboulen Thanks for letting me know.  I looked at all my test courses that I created for this and couldn't see a second image.

Logically perhaps it would make sense for the first course image to overwrite the second, but then the intent of the action is the content of the course and the image is a display thing.  But as this is not a clear answer as some people will want the image overwritten and others not, then I'm going to leave as is.  To overwrite the image will lead to even more complexity.
